blocktx/PeerHandler.go

```diff
@@ -352,21 +361,42 @@ func (bs *PeerHandler) insertBlock(blockHash *chainhash.Hash, merkleRoot *chainh
 	return bs.store.InsertBlock(context.Background(), block)
 }
 
-func (bs *PeerHandler) markTransactionsAsMined(blockId uint64, transactionHashes []*chainhash.Hash) error {
+func (bs *PeerHandler) markTransactionsAsMined(blockId uint64, transactionHashes []*chainhash.Hash, merkleTree []string) error {
 	start := gocore.CurrentNanos()
 	defer func() {
 		gocore.NewStat("blocktx").NewStat("HandleBlock").NewStat("markTransactionsAsMined").AddTime(start)
 	}()
 
 	txs := make([]*blocktx_api.TransactionAndSource, 0, len(transactionHashes))
+	merklePaths := make([]string, 0, len(transactionHashes))
 
-	for _, hash := range transactionHashes {
+	for txIndex, hash := range transactionHashes {
 		txs = append(txs, &blocktx_api.TransactionAndSource{
 			Hash: hash[:],
 		})
+
+		var merklePath string
+		var err error
+		if len(transactionHashes) != 1 {
+			merklePath, err = bc.GetTxMerklePath(txIndex, merkleTree).String()
+			if err != nil {
+				return err
+			}
+
+			root, err := bc.GetTxMerklePath(txIndex, merkleTree).CalculateRoot(hash.String())
+			if err != nil {
+				return err
+			}
+			fmt.Println(root)
```

Comment:
Was this printing used for debugging?
If not, rather use the logger to print it.